openlibrary/tests/test_templates.py

```diff
@@ -0,0 +1,25 @@
+from web.template import Template
+from tokenize import TokenError
+
+
+def try_parse_template(template_text, filename=''):
+    """
+    :param str template_text:
+    :param str filename:
+    :rtype: (bool, str or Exception)
+    """
+    try:
+        text = Template.normalize_text(template_text)
```

Comment:
I don't think this is required, web.py runs the text through `normalize_text()` already: 
https://github.com/webpy/webpy/blob/v0.39/web/template.py#L843-L846

Reply:
Oh, that's odd; I was getting some odd failures when I didn't run this step :/ I'll investigate.